src/Mod/Sketcher/Gui/DrawSketchDefaultWidgetController.h

```diff
@@ -253,25 +253,7 @@ class DrawSketchDefaultWidgetController: public DrawSketchController<HandlerT,
 
     /// Automatic default method update in combobox
     void doConstructionMethodChanged() override
-    {
-        nParameter = WidgetParametersT::size(handler->constructionMethod());
-        nCheckbox = WidgetCheckboxesT::size(handler->constructionMethod());
-        nCombobox = WidgetComboboxesT::size(handler->constructionMethod());
-
-        // update the combobox only if necessary (if the change was not triggered by the
-        // combobox)
-        if constexpr (PFirstComboboxIsConstructionMethod == true) {
-            auto currentindex = toolWidget->getComboboxIndex(WCombobox::FirstCombo);
-            auto methodint = static_cast<int>(handler->constructionMethod());
-
-            if (currentindex != methodint) {
-                // avoid triggering of method change
-                boost::signals2::shared_connection_block combobox_block(
-                    connectionComboboxSelectionChanged);
-                toolWidget->setComboboxIndex(WCombobox::FirstCombo, methodint);
-            }
-        }
-    }
+    {}

```

Comment:
I am curious. 

This you moved to the reset function below. However, I cannot come to a reason why is this change necessary.

 In other words, do we need this code on every reset, even when the construction mode did not change? Is it not only necessary when changing construction mode?

Reply:
This is commit  [da256f2](https://github.com/FreeCAD/FreeCAD/pull/11294/commits/da256f2a94b76395195e19e59f74c5b6f67f36da)
It is to solve two problems. When you initialize a tool with a construction methode that is not default, then you have 2 problems : 
1 - The number of parameters / checkbox / combobox / OVP initialized is not the correct one. Which results in crash for Arc tool when you try to initiate 3 rim point arc.
2 - The combobox do not show the correct entry. Currently in master if you start the command 'centered rectangle' then the combobox shows 'diagonal'.

Reply:
So if we keep the multiple tools in the toolbars (3 rim points arcs, centered rectangle, oblong) this is necessary.
But if we remove the tools, then its not necessary anymore.

Reply:
Ah! It is the overriding of the default construction mode on initialisation. Good then! Thanks.